assets/prototype/application/ui/components/forms/formModal/types.ts

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { FormProps, FormRenderProps } from 'react-final-form';
+import { FormApi } from 'final-form';
+
+export interface FormModalProps extends FormProps {
+	FormComponent: React.FC;
+	onClose: (e?: any) => void;
+	isOpen: boolean;
+}
+
+export interface FormComponentProps {
+	id?: string;
+	title?: string;
+	form?: FormApi;
+	values?: any;
+	submitting?: boolean;
+	pristine?: boolean;
+	formReset?: boolean;
+}
```

Comment:
all optional? can this not be tightened up? I would think some of those are required, no?

Reply:
None of those is present in every modal form :)

Reply:
wow! ok... would have thought form and id were used consistently... ok